docs/developer-guide/typeConversions.md

```diff
@@ -0,0 +1,211 @@
+
+#### getIntFromString
+
+Converts a string into 64 bit integer. If the string has properties of a double or vector it will truncate the values from `getDoubleFromString`.
+
+#### getDoubleFromString
+
+Converts a string into a double value, a complex, or a vector of real or complex numbers. If the vector is more than a single element the output is the vector norm of the vector. If the string is not convertible the invalid_double is returned (-1e49).
+
+#### helicsGetComplexFromString
+
+Similar to getDoubleFromString in conversion of vectors. It will convert most representations of complex number patterns using a trailing `i` or `j` for the imaginary component and assumes the imaginary component is last. The real component can be omitted if not present. For Example `4.7+2.7j`or`99.453i`
+
+#### helicsBoolValue
+
+"0", "00","0000", "\0","false","f","F","FALSE","N","no","n","OFF","off","","disable","disabled" all return false everything else return true
+
+### Conversion from vector double
+
+- INT -> trunc(vectorNorm(val)) [^1]
+- DOUBLE -> vectorNorm(val)
+- COMPLEX -> val[0]+val[1]j
+- VECTOR - > val
+- COMPLEX_VECTOR -> [val[0],val[1],...,val[N]]
+- STRING -> vectorString(val) [^2]
+- NAMED_POINT ->{vectorString(val), NAN} [^3]
+- BOOL ->(vectorNorm(val)!=0)?"1":"0"
+
+[^1]: vectorNorm is the sqrt of the inner product of the vector.
+[^2]: vectorString is comma separated string of the numerical values enclosed in `[]` for example `[45.7,22.7,17.8]` this is a JSON compatible string format.
+[^3]: if the vector is a single element the NAMED_POINT translation is equivalent to a double translation.
+
+### Conversion from Complex Vector
+
+- INT -> trunc(vectorNorm(val))
+- DOUBLE -> vectorNorm(val)
+- COMPLEX -> val[0]
+- VECTOR - > [abs(val[0]),abs(val[1]),...abs(val[N])][^1]
+- COMPLEX_VECTOR -> val
+- NAMED_POINT ->{vectorString(val), NAN}
+- STRING -> complexVectorString(val)
+- BOOL ->(vectorNorm(val)!=0)?"1":"0"
+
+See Conversion from double for definitions of vector Norm
+[^1]: if the imaginary part of these values is 0, then the real part is used otherwise it uses the absolute value;
+
+### Conversion from Complex
+
+- INT -> trunc((val.imag() == 0)?val.real(): std::abs(val))
+- DOUBLE -> val.imag() == 0)?val.real(): std::abs(val)
+- COMPLEX -> val
+- VECTOR - > [val.real,val.imag]
+- COMPLEX_VECTOR -> [val]
+- NAMED_POINT ->{helicsComplexString(val), NAN}
+- STRING -> helicsComplexString(val)
+- BOOL ->(std::abs(val)!=0)?"1":"0"
+
+If the imaginary value == 0, the value is treated the same as a double
+
+### Conversion from a NamedPoint
+
+if the value of the named point is a `NAN` treat the name part the same as a string
+otherwise use the numerical value as a double and convert appropriately. The exception is a string which has a dedicated operation to generate a JSON string with two fields {"name" and "value"}
```

Comment:
```suggestion
If the value of the named point is `NAN` then treat the name part the same as a string,
otherwise use the numerical value as a double and convert appropriately. The exception is a string which has a dedicated operation to generate a JSON string with two fields {"name" and "value"}.
```